server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java

```diff
@@ -83,7 +85,15 @@ public final void start(A configuration, EmbeddedCacheManager cacheManager) {
    }
 
    protected void startTransport() {
-      InetSocketAddress address = new InetSocketAddress(configuration.host(), configuration.port());
+      int port = configuration.port();
```

Comment:
I don't think that this code is needed. From the `InetSocketAddress` javadoc:
```
 A port number of zero will let the system pick up an ephemeral port in a bind operation
```
https://docs.oracle.com/javase/8/docs/api/java/net/InetSocketAddress.html#InetSocketAddress-java.net.InetAddress-int-